src/test/java/com/arpnetworking/metrics/common/sources/ActorSourceTest.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2016 Smartsheet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.common.sources;
+
+import akka.actor.ActorSystem;
+import com.arpnetworking.metrics.mad.actors.SourceSupervisor;
+import org.junit.After;
+import org.junit.Before;
+import scala.concurrent.Await;
+import scala.concurrent.duration.Duration;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test base for actor sources.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public abstract class ActorSourceTest {
```

Comment:
I would also rename this to something like BaseActorSourceTest or some such. This is also the third time I've seen similar functionality. Time to refactor into commons I think. Also, the one problem I've seen with all of them is lack of ability to specify a custom configuration file for the ActorSystem.